lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -335,6 +346,7 @@ public function errorInfo()
     public function closeCursor()
     {
         $this->_stmt->free_result();
+        $this->_rowValuesAreBound = false;
```

Comment:
@kimhemsoe by "reset" you mean re-bind? We can extract the procedure of binding row values to a separate method and call it from `execute()` and from here.